### PR TITLE
Bug 1823367 - When closing Fenix in private mode, the ETP setting should have the default value.

### PR DESCRIPTION
**Pull request by iorgamgabriel**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1823367

### CODE REVIEW

**Comment by iorgamgabriel** [comment]
https://github.com/mozilla-mobile/firefox-android/assets/8118371/54da7f53-78d4-49bd-ab05-263727f59a1e

File: fenix/app/src/main/java/org/mozilla/fenix/session/VisibilityLifecycleCallback.kt [hunk 1]
```diff
@@ -62,6 +67,23 @@ class VisibilityLifecycleCallback(private val activityManager: ActivityManager?)
...
+    private fun removeTCPException(activity: Activity) {
+        val privateTabs = activity.components.core.store.state.privateTabs
+        for (privateTab in privateTabs) {
+            val trackingProtectionUseCases =
+                activity.components.useCases.trackingProtectionUseCases
+            trackingProtectionUseCases.containsException(privateTab.id) {
+                if (it) {
+                    trackingProtectionUseCases.removeException(privateTab.id)
+                }
+            }
```
Comment (Amejia481) [inline fenix/app/src/main/java/org/mozilla/fenix/session/VisibilityLifecycleCallback.kt:83]: As this is doing a round trip to gecko, we could avoid doing that by query `tab.trackingProtection.ignoredOnTrackingProtection` which is set every time we [add/remove](https://github.com/mozilla-mobile/firefox-android/blob/d88fcfa9c409459bf5a7b5e83d56c3bd3c394879/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoTrackingProtectionExceptionStorage.kt#L71) a TP exception for a session.

>         activity.components.core.store.state.privateTabs.filter {
>             it.trackingProtection.ignoredOnTrackingProtection
>         }.forEach { privateTab ->
>             activity.components.useCases.trackingProtectionUseCases.removeException(privateTab.id)
>         }

**Review by Amejia481 — APPROVED** [review]
Thanks for the patch, I added a comment below.

File: fenix/app/src/main/java/org/mozilla/fenix/session/VisibilityLifecycleCallback.kt (hunk 1 above)
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/session/VisibilityLifecycleCallback.kt]: Done

**Pull request merged**